enterprise_access/apps/content_assignments/tests/test_tasks.py

```diff
@@ -0,0 +1,167 @@
+            {
+                'enterprise_customer': str(TEST_ENTERPRISE_UUID),
+                'user_email': TEST_EMAIL,
+            },
+            response_status_that_triggers_retry,
+        )
+
+        task_result = create_pending_enterprise_learner_for_assignment_task.delay(self.assignment.uuid)
+
+        # Celery thinks the task failed.
+        assert task_result.state == celery_states.FAILURE
+
+        # The overall task result is just the HTTPError bubbled up from the API response.
+        assert isinstance(task_result.result, HTTPError)
+        assert task_result.result.response.status_code == response_status_that_triggers_retry
+
+        # The LMS/enterprise API was called once plus the max number of retries, all with the correct request body.
+        assert len(mock_oauth_client.return_value.post.call_args_list) == 1 + settings.TASK_MAX_RETRIES
+        for call in mock_oauth_client.return_value.post.call_args_list:
+            assert call.kwargs['json'] == [{
+                'enterprise_customer': str(self.assignment.assignment_configuration.enterprise_customer_uuid),
+                'user_email': self.assignment.learner_email,
+            }]
+
+        self.assignment.refresh_from_db()
+        assert self.assignment.state == LearnerContentAssignmentStateChoices.ERRORED
+
+    @mock.patch('enterprise_access.apps.api_client.base_oauth.OAuthAPIClient')
+    def test_last_retry_success(self, mock_oauth_client):
+        """
+        Test a scenario where the API response keeps triggering a retry until the last attempt, then finally responds
+        successfully.
+        """
+        # Mock multiple consecutive responses, only the last of which was successful.
+        retry_triggering_responses = [
+            MockResponse(None, status.HTTP_503_SERVICE_UNAVAILABLE)
+            for _ in range(settings.TASK_MAX_RETRIES)
+        ]
+        final_success_response = MockResponse(
+            {
+                'enterprise_customer': str(TEST_ENTERPRISE_UUID),
+                'user_email': TEST_EMAIL,
+            },
+            status.HTTP_201_CREATED,
+        )
+        mock_oauth_client.return_value.post.side_effect = retry_triggering_responses + [final_success_response]
+
+        task_result = create_pending_enterprise_learner_for_assignment_task.delay(self.assignment.uuid)
+
+        # Celery thinks the task succeeded.
+        assert task_result.state == celery_states.SUCCESS
+
+        # The LMS/enterprise API was called once plus the max number of retries, all with the correct request body.
+        assert len(mock_oauth_client.return_value.post.call_args_list) == 1 + settings.TASK_MAX_RETRIES
+        for call in mock_oauth_client.return_value.post.call_args_list:
+            assert call.kwargs['json'] == [{
+                'enterprise_customer': str(self.assignment.assignment_configuration.enterprise_customer_uuid),
+                'user_email': self.assignment.learner_email,
+            }]
```

Comment:
could we get this assertion down here, too?
```python
self.assignment.refresh_from_db()
assert self.assignment.state == LearnerContentAssignmentStateChoices.ALLOCATED
```

Reply:
No harm in adding that, but just to be clear the happy path doesn't alter the assignment state, so there's no testable change.

Reply:
Oh, I looked at the test name and realized you're actually just testing that it doesn't change to errored. Gotcha!